utility_analysis_new/histograms.py

```diff
@@ -229,13 +305,20 @@ def compute_contribution_histograms(
     # col: (pid, pk)
 
     col = backend.to_multi_transformable_collection(col)
-    cross_partition_histogram = _compute_cross_partition_histogram(col, backend)
+    cross_partition_histogram = _compute_l0_contributions_histogram(
+        col, backend)
     # 1 element collection: ContributionHistogram
-    per_partition_histogram = _compute_per_partition_histogram(col, backend)
+    per_partition_histogram = _compute_linf_contributions_histogram(
+        col, backend)
+
+    partition_count_histogram = _compute_partition_count_histogram(col, backend)
+    partition_privacy_id_count_histogram = _compute_partition_privacy_id_count_histogram(
```

Comment:
I feel like we are doing very similar computations to come up with these 4 histograms. Maybe we can refactor a bit to get rid of unnecessary computation?

That said, this might not be a bottleneck, in which case we could ignore this.